smartsim/_core/mli/infrastructure/control/device_manager.py

```diff
@@ -66,15 +66,23 @@ def remove_model(self, key: str) -> None:
 
         :param key: The key of the model to remove
         """
-        self._models.pop(key)
+        try:
+            self._models.pop(key)
+        except KeyError:
+            logger.warning(f"An unknown key was requested for removal: {key}")
+            raise
 
     def get_model(self, key: str) -> t.Any:
         """Get the model corresponding to a given key.
 
         :param key: The model key
         :returns: The model for the given key
         """
-        return self._models[key]
+        try:
+            return self._models[key]
+        except KeyError:
+            logger.warning(f"An unknown key was requested: {key}")
```

Comment:
Shouldn't we raise this?

Reply:
(or let the function throw an Exception)

Reply:
I have a `raise` under the `logger.warning`. Do you mean like raise a `KeyError` with a different message or something?

Reply:
Oh, I think I missed the `raise` line! Should we add `:raises KeyError:` to the docstring?

Reply:
(same comment applies for `remove_model`)

Reply:
fixed x2!